source/agora/common/TransactionPool.d

```diff
@@ -0,0 +1,248 @@
+/*******************************************************************************
+
+    Contains a transaction pool that is serializable to disk,
+    using SQLite as a backing store.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.TransactionPool;
+
+import agora.common.Data;
+import agora.common.Deserializer;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.common.Transaction;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+import vibe.core.log;
+
+import std.algorithm;
+import std.conv : to;
+import std.exception : collectException, enforce;
+import std.file : exists;
+import std.range;
+
+/// Ditto
+public class TransactionPool
+{
+    // should only be initialized once per process
+    shared static this()
```

Comment:
Is a `shared static this` the only way to do this ? (shared) module ctor have quite a lot of drawback / friction with other features.

Reply:
Hmm not necessarily. We could do it in the main module and also the main test runner module.